src/main/java/edu/kis/powp/jobs2d/command/ImmutableCompoundCommand.java

```diff
@@ -11,7 +12,9 @@
     private List<DriverCommand> driverCommands = null;
 
     public ImmutableCompoundCommand(List<DriverCommand> driverCommands) {
-        this.driverCommands = Collections.unmodifiableList(driverCommands);
+        this.driverCommands = new ArrayList<>();
+        this.driverCommands.addAll(driverCommands);
+        this.driverCommands = Collections.unmodifiableList(this.driverCommands);
```

Comment:
This doesn't change much. Cooperate with the group responsible for "A deep copy of a complex command" to fix the solution.

Reply:
Corrected deep copying in latest commit.